### PR TITLE
os/dbuild.sh : Refactoring of build and download script

### PR DESCRIPTION
**Pull request by hasw7569**
I changed the hardcoded part of the build script to read board-specific predefined options.
In addition, I added board-specific options to provide the user with a menu

I also refactored download scripts

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [ff2d8f2b7080d59c94ee1b73219d9ca26579ab94] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [ff2d8f2b7080d59c94ee1b73219d9ca26579ab94] - Code Rule Check (C++) OK.

**Comment by jeongchanKim** [comment]
Please fix the commit title : .h -> .sh

**Comment by seinfra** [comment]
Target : [082a82cee4ca9f96ad4d4324badc8d3576a6c501] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [082a82cee4ca9f96ad4d4324badc8d3576a6c501] - Code Rule Check (C++) OK.

File: build/configs/artik053/.flashSpec.xml [hunk 1]
```diff
@@ -23,11 +23,18 @@
...
         <option desc="Register the USB rule">USBrule</option>
-        <option desc="Flash Bootloader binaries">BL1 BL2</option>
-        <option desc="Flash Firmware binaries">SSSFW WLANFW</option>
-        <option desc="Flash ROMFS image">ROMFS</option>
-        <option desc="Erase OTA partition">ERASE_OTA</option>
-        <option desc="Erase user file system">ERASE_USERFS</option>
-        <option desc="Erase all of partitions">ERASE_ALL</option>
+        <option desc="Flash TizenRT APPS">APPS</option>
+        <option desc="Flash TizenRT OTA">OTA</option>
+        <option desc="Flash TizenRT ROMFS">ROM</option>
```
Comment (jeongchanKim) [inline build/configs/artik053/.flashSpec.xml]: I think ROMFS is much better.

File: build/configs/artik05x/artik05x_download.sh [hunk 2]
```diff
@@ -273,25 +272,40 @@ while test $# -gt 0; do
...
-			usage 1>&2
-			exit 1
+			break
 			;;
 	esac
 	shift
 done
+
+case $1 in
+	ALL|OS|APPS|ROM|BL1|BL2|SSSFW|WLANFW|OTA|MICOM|WIFI|USER|all|os|apps|rom|bl1|bl2|sssfw|wlanfw|ota|micom|wifi|user)
```
Comment (jeongchanKim) [inline build/configs/artik05x/artik05x_download.sh]: How about not distinguising the upper and lower case?

Comment (hasw7569) [reply build/configs/artik05x/artik05x_download.sh]: I agree with your opinion.
I will change this.

**Comment by seinfra** [comment]
Target : [5081f002f34b0358c9b30eef3e0f5955e8f7cf88] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [5081f002f34b0358c9b30eef3e0f5955e8f7cf88] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [eb3a77eddb0fbfc4fa8e5521fb546b16b602e7aa] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [eb3a77eddb0fbfc4fa8e5521fb546b16b602e7aa] - Code Rule Check (C++) OK.

File: build/configs/artik05x/artik05x_download.sh (hunk 2 above)
Comment (hasw7569) [reply build/configs/artik05x/artik05x_download.sh]: modified

**Comment by seinfra** [comment]
Target : [b96fdfcb5ab74225b8f6a6279058a71775f1def5] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [b96fdfcb5ab74225b8f6a6279058a71775f1def5] - Code Rule Check (C++) OK.

File: os/dbuild.sh [hunk 3]
```diff
@@ -294,40 +294,78 @@ function SELECT_CONFIG()
...
 		echo "  \"x. Exit\""
 		echo ==================================================
 		read SELECTED_DL
 	fi
 
+	for ((i=1;i<=${#command[@]};i++)); do
+		if [ $SELECTED_DL == $i ]; then
+			DL_ARG=${command[$i-1]}
+		fi
+	done
```
Comment (sunghan-chang) [inline os/dbuild.sh]: Is this working with below?
```
./dbuild.sh download os
```

Comment (hasw7569) [reply os/dbuild.sh]: Yes It works well

**Comment by seinfra** [comment]
Target : [c30f05f3cd3911702751911e97b27eba66c04800] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [c30f05f3cd3911702751911e97b27eba66c04800] - Code Rule Check (C++) OK.

File: build/configs/artik05x/artik05x_download.sh [hunk 4]
```diff
@@ -273,20 +276,25 @@ while test $# -gt 0; do
...
 			;;
-		ALL|OS|APPS|ROM|BL1|BL2|SSSFW|WLANFW|OTA|MICOM|WIFI|USER|all|os|apps|rom|bl1|bl2|sssfw|wlanfw|ota|micom|wifi|user)
-			download $1
+		all|os|apps|rom|bl1|bl2|sssfw|wlanfw|ota|micom|wifi|user)
+			download ${1,,}
 			;;
-		USBrule|usbrule)
-			RET=$(${THIS_PATH}/../usbrule.sh artik 0403 6010)
+		usbrule)
+			RET=`${USBRULE_PATH} ${USBRULE_BOARD} ${USBRULE_IDVENDOR} ${USBRULE_IDPRODUCT}`
```
[the remark this answers is not in the document]
Comment (hasw7569) [reply build/configs/artik05x/artik05x_download.sh]: I think your opinion is better
I will change this.

File: build/configs/artik05x/artik05x_download.sh [hunk 5]
```diff
@@ -273,20 +276,25 @@ while test $# -gt 0; do
...
 			if [ -n "${RET}" ]; then
 				echo $RET
 				exit 1
 			fi
 			echo "USB rule creation succeeded."
 			exit 0
```
[the remark this answers is not in the document]
Comment (hasw7569) [reply build/configs/artik05x/artik05x_download.sh]: I agree.
I will remove unnecessary line

File: build/configs/artik05x/artik05x_download.sh (hunk 4 above)
[the remark this answers is not in the document]
Comment (hasw7569) [reply build/configs/artik05x/artik05x_download.sh]: modified

File: build/configs/artik05x/artik05x_download.sh (hunk 5 above)
[the remark this answers is not in the document]
Comment (hasw7569) [reply build/configs/artik05x/artik05x_download.sh]: modified

File: build/configs/artik05x/artik05x_download.sh [hunk 6]
```diff
@@ -273,20 +276,25 @@ while test $# -gt 0; do
...
 			;;
-		ERASE_*)
-			erase
+		erase)
+			while test $# -gt 1
+			do
+				optarg=${2,,}
+				erase
```
[the remark this answers is not in the document]
Comment (hasw7569) [reply build/configs/artik05x/artik05x_download.sh]: modified

**Comment by seinfra** [comment]
Target : [ebe1901279fd1caca5b0c27272e1a274ffd513e0] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [ebe1901279fd1caca5b0c27272e1a274ffd513e0] - Code Rule Check (C++) OK.

**Pull request merged**